### DISCUSSION TITLE
Changing size of sprites in spritesheet bundle

### DISCUSSION THREAD

**Discussion by gmoller** [bevyengine/bevy#2840] [Q&A] [answered]
Hi, I have a spritesheet with large sprites (256x384). I want to render these sprites to the screen as a smaller size (e.g. 64x96).

I have written a function to create the bundles:
fn create_spritesheet_bundle(position: Vec3, scale: Vec3, texture_atlas_handle: &Handle<TextureAtlas>, sprite_index: u32) -> 
SpriteSheetBundle {

    let sprite = TextureAtlasSprite::new(sprite_index);
    let transform = Transform {
        translation: position,
        scale: **scale**,
        ..Default::default()
    };

    let spritesheet_bundle = SpriteSheetBundle {
        texture_atlas: texture_atlas_handle.clone(),
        sprite: sprite,
        transform: transform,
        ..Default::default()
    };

    spritesheet_bundle
}

I can adjust the size by using the Scale, but was wondering if there is a way to do it like I can with my normal sprite creation function:
fn create_sprite_bundle(size: Vec2, position: Vec3, scale: Vec3, material: Handle<ColorMaterial>) -> SpriteBundle {

    let sprite = **Sprite::new(size);**
    let transform = Transform {
        translation: position,
        scale: scale,
        ..Default::default()
    };

    let sprite_bundle = SpriteBundle {
        material: material,
        sprite: sprite,
        transform: transform,
        ..Default::default()
    };

    sprite_bundle
}

I hope my question makes sense.

Thanks.

**Comment by VVishion** [comment, marked as answer]
I want to change the bundle for `TextureAtlas`es a bit once the renderer rework lands on main. If a `Sprite` is constructed with a Texture its bundle contains `Sprite`, `Handle<Image>` etc. If it's constructed from a `TextureAtlas`-Texture its bundle contains `TextureAtlasSprite`, `Handle<TextureAtlas>` etc. I want to unify this. The Sprite Texture will be represented by:
* for the case of a Texture: `Handle<Image>`,
* for the case of a Texture inside a `TextureAtlas`: `Handle<TextureAtlas>` + `TextureAtlasEntry`,

Both cases will be bundled with `Sprite`. So no more `TextureAtlasSprite` and unified functionality.
Then, you can use the `custom_size` property of `Sprite` for both cases. Feel free to create an Issue for this.
But for now you have to use your little hack.

**Reply by gmoller** [reply]
Thanks for the quick reply. Yeah, I agree it would be nice if they were unified. I'm going to play around with using individual png's for my tiles and see how that works out.
